include/psa/crypto_values.h

```diff
@@ -2083,10 +2165,38 @@ static inline int mbedtls_svc_key_id_is_null( mbedtls_svc_key_id_t key )
  */
 #define PSA_KEY_USAGE_VERIFY_HASH               ((psa_key_usage_t)0x00002000)
 
-/** Whether the key may be used to derive other keys.
+/** Whether the key may be used to derive other keys or produce a password
+ * hash.
+ *
+ * This flag allows the key to be used as the input of
+ * psa_key_derivation_input_key() at the step
+ * #PSA_KEY_DERIVATION_INPUT_SECRET of #PSA_KEY_DERIVATION_INPUT_PASSWORD
+ * depending on the algorithm, and allows the use of
+ * psa_key_derivation_output_bytes() or psa_key_derivation_output_key()
```

Comment:
This is not addressed. Strictly speaking the language here doesn't claim that the list of functions is exhaustive, but it's a bit misleading not to list the verify functions.

Reply:
Well, actually in the current version of this PR, the verify functions need the `VERIFY_DERIVATION` bit, as documented in the error section of these functions and in the note on `input_key()`. I don't have a strong opinion on it remaining that way, but let me share the reasoning that led to the current state so that we can assess it and decide whether we change it or not.

If the `DERIVE` bit allows everything, people might be tempted to use it everywhere for simplicity and not think whether `VERIFY_DERIVATION` would suffice for what they're doing. Also (or perhaps it's just another way of saying the same thing?), it seemed a bit more orthogonal/consistent to me for each set of functions (output, derive) to be independently controlled by its own usage bit, rather than one usage bit being a superset of the other.

Again, I'm not married to this line of reasoning, just spelling out my thought process so that we can discuss it.

Reply:
Thinking about it again, I agree with your reasoning. Looking at derivation alone, it doesn't really make sense to allow output() without allowing verify(). However, it's just up to the application not to do that. For other operations, we decided to keep things simple and have one permission flag map to one operation, and for example you can't verify a MAC with `USAGE_VERIFY` alone even though you can calculate the MAC and compare it. So key derivation should follow the same principle.

There is however one reason to make derivation different, which is backward compatibility. Suppose I've created a key derivation key with an implementation of PSA Crypto 1.0.0, and it has `USAGE_DERIVE` because that's all there is. It happens that my application calls `output_bytes` only to then compare these bytes with a reference value. So now I want to call `verify_bytes` instead, just to simplify my code, but I can't.

I'm not convinced that this is a compelling reason, so let's go for the simple way now (no change).